spock-specs/src/test/groovy/org/spockframework/EmbeddedSpecification.groovy

```diff
@@ -30,8 +32,21 @@ abstract class EmbeddedSpecification extends Specification {
   EmbeddedSpecRunner runner = new EmbeddedSpecRunner()
   EmbeddedSpecCompiler compiler = new EmbeddedSpecCompiler()
 
+  def javaVersion = System.getProperty("java.specification.version") as BigDecimal
+
   void stackTraceLooksLike(Throwable exception, String template) {
-    def trace = exception.stackTrace
+
+    def trace
+
+    if (javaVersion >= 9) {
+      trace = Arrays.stream(exception.getStackTrace())
```

Comment:
can be done with groovy functions without requiring Java8

Reply:
Fixed.